swift/apple/FirezoneNetworkExtension/NetworkSettings.swift

```diff
@@ -141,7 +149,8 @@ class NetworkSettings {
 
     let dnsSettings = NEDNSSettings(servers: [dnsAddress])
     // Intercept all DNS queries; SplitDNS will be handled by connlib
-    dnsSettings.matchDomains = [""]
+    dnsSettings.matchDomains = matchDomains
+    dnsSettings.matchDomainsNoSearch = true
```

Comment:
Prevents the dummy matchDomain from appearing in /etc/resolv.conf

Reply:
Agreed. `matchDomainsNoSearch` should have been set irrespective of this libresolv workaround.